gdx/src/com/badlogic/gdx/math/Rectangle.java

```diff
@@ -303,6 +303,60 @@ public float getAspectRatio () {
 		return (height == 0) ? Float.NaN : width / height;
 	}
 
+	/** Returns the bottommost coordinate of the rectangle.
+	 * @return The bottommost coordinate of the rectangle. */
+	public float getBottom () {
+		return y;
+	}
+
+	/** Sets the Y coordinate so that the bottom of the rectangle is at the given value.
+	 * @param y The value for the bottom. */
+	public void setBottom (float y) {
+		this.y = y;
+	}
+
+	/** Returns the topmost coordinate of the rectangle.
+	 * @return The topmost coordinate of the rectangle. */
+	public float getTop () {
+		return y + height;
+	}
+
+	/** Sets the Y coordinate so that the top of the rectangle is at the given value.
+	 * @param y The value for the top. */
+	public void setTop (float y) {
+		this.y = y - height;
+	}
+
+	/** Returns the rightmost coordinate of the rectangle.
+	 * @return The rightmost coordinate of the rectangle. */
+	public float getRight () {
+		return x + width;
+	}
+
+	/** Sets the X coordinate so that the right of the rectangle is at the given value.
+	 * @param x The value for the right. */
+	public void setRight (float x) {
+		this.x = x - width;
+	}
+
+	/** Returns the leftmost coordinate of the rectangle.
+	 * @return The leftmost coordinate of the rectangle. */
+	public float getLeft () {
+		return x;
+	}
+
+	/** Sets the X coordinate so that the left of the rectangle is at the given value.
+	 * @param x The value for the left. */
+	public void setLeft (float x) {
+		this.x = x;
+	}
+
+	/** Calculates the center of the rectangle.
+	 * @return A vector with the center position. */
+	public Vector2 getCenter () {

```

Comment:
@varomorf this one is generating garbage because you're creating a new object each time you call it. Ideally in your game code, you should have a single Vector2 property that you re-use to pass to getCenter().